File: WordsManager/src/main/java/com/example/WordsManager/models/Word.java
package com.example.WordsManager.models;


import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.io.Serializable;
import java.util.Comparator;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "word")
public class Word implements Serializable, Comparator<Word> {
    @Id
    @Column("id")
    private int id;
    @Column("foreign_word")
    private String foreignWord;
    @Column("transcription")
    private String transcription;
    @Column("translation")
    private String translation;
    @Column("description")
    private String description;
    @Column("link_voice")
    private String link_voice;
    @Column("link_image")
    private String link_image;
    @Column("topic")
    private String topic;
    @Column("sorting_value")
    private Integer sorting_value;


    @Override
    public int compare(Word o1, Word o2) {
        float f = o1.sorting_value-o2.sorting_value;
        if (f>0)return 1;
        else if (f<0)return -1;
        else return 0;
    }
}

File: ZubrilkaEnglish(android)/app/src/main/java/com/zubrilka/zubrilkaenglish/models/Word.kt
package com.zubrilka.zubrilkaenglish.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "all_word_table")
data class Word(
    @PrimaryKey(autoGenerate = false)
    val id: Int,
    @ColumnInfo(name = "foreignWord")
    val foreignWord:String?,
    @ColumnInfo(name = "transcription")
    val transcription:String?,
    @ColumnInfo(name = "translation")
    val translation:String?,
    @ColumnInfo(name = "description")
    val description:String?,
    @ColumnInfo(name = "topic")
    val topic:String,
    @ColumnInfo(name = "link_voice")
    val link_voice: String?,
    @ColumnInfo(name = "link_image")
    val link_image: String?,
    @ColumnInfo(name = "sorting_value")
    val sorting_value: Int
){
    override fun toString(): String {
        return "Word $id, $foreignWord, $transcription, $translation, $description, $topic, $link_voice, $link_image"
    }
}